src/redis-cli.c

```diff
@@ -3968,6 +3968,15 @@ static int clusterManagerMoveSlot(clusterManagerNode *source,
     if (!success) return 0;
     /* Set the new node as the owner of the slot in all the known nodes. */
     if (!option_cold) {
+        /* If source node lost last slot and target node receives CLUSTER SETSLOT
+         * before source node, source node will be replica of target node. Then
+         * source node will report error when CLUSTER SETSLOT is received. So we
+         * need to send CLUSTER SETSLOT to source node before target node. See
+         * https://github.com/redis/redis/issues/9223 */
```

Comment:
This is a dangerous idea. See #7116.

If the target node crashes when the source node is no longer the slot owner but before the destination is the owner, then the slot has no owner.